Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/spark/run/SparkBatchJobRemoteDebugProcess.java

```diff
@@ -0,0 +1,138 @@
+            return SparkBatchRemoteDebugJob.factory(
+                    URI.create(JobUtils.getLivyConnectionURL(cluster)).toString(),
+                    getSubmissionParameter(),
+                    SparkBatchSubmission.getInstance());
+        } catch (DebugParameterDefinedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+
+    @Override
+    protected Observable<SimpleImmutableEntry<SparkBatchJobState, String>> awaitForJobDone(SparkBatchJob runningJob) {
+        return createDebugSession((SparkBatchRemoteDebugJob) runningJob)
+                .subscribeOn(getSchedulers().processBarVisibleAsync("Create Spark batch job debug session"))
+                .flatMap(super::awaitForJobDone);
+    }
+
+    @NotNull
+    protected SparkBatchDebugJobJdbPortForwardedEvent createEventWithJdbPorForwarding(SparkBatchRemoteDebugJob job)
+            throws SparkJobException, JSchException, IOException {
+        SparkBatchDebugSession session = SparkBatchDebugSession.factoryByAuth(job.getConnectUri().toString(),
+                                                                              getAuthData())
+                                                               .open();
+
+        String remoteHost = job.getSparkDriverHost();
+        int remotePort = job.getSparkDriverDebuggingPort();
+
+        int localPort = session
+                .forwardToRemotePort(remoteHost, remotePort)
+                .getForwardedLocalPort(remoteHost, remotePort);
+
+        // Start to find executors
+        executorSubscription = job.getExecutorsObservable()
+                .map(hostContainerPair -> new SparkBatchJobExecutorCreatedEvent(
+                        job, session, hostContainerPair.getKey(), hostContainerPair.getValue()))
+                .subscribe(getEventSubject()::onNext);
+
+        return new SparkBatchDebugJobJdbPortForwardedEvent(job, session, remoteHost, remotePort, localPort, true);
+    }
+
+    @Override
+    public void disconnect() {
+        super.disconnect();
+
+        if (executorSubscription != null) {
+            executorSubscription.unsubscribe();
+        }
+    }
+
+    private Observable<SparkBatchRemoteDebugJob> createDebugSession(SparkBatchRemoteDebugJob job) {
+        return Observable.fromCallable(() -> {
+            try {
+                SparkBatchDebugJobJdbPortForwardedEvent jdbReadyEvent = createEventWithJdbPorForwarding(job);
+
+                // Debug session created and SSH port forwarded
+                getEventSubject().onNext(jdbReadyEvent);
+
+                return job;
+            } catch (JSchException e) {
+                // Rethrow it since JSch can't handle the certificate expired issue
+                throw new SparkJobException(
+                        "Can't create Spark Job remote debug session, " +
+                                "please check your SSH connect with manually login.",
+                        e);
```

Comment:
As discussed, can we check whether SSH password expired? Or at least add clear message, "Please check whether SSH password has expired, using Putty or other SSH tool"